apps/github-cascading-app/src/cascading/cascading-probot.ts

```diff
@@ -0,0 +1,230 @@
+import { ProbotOctokit } from 'probot';
+import { BaseLogger, CascadingConfiguration, CascadingPullRequestInfo, CheckConclusion } from './interfaces';
+import { Cascading } from './cascading';
+
+export interface CascadingProbotOptions {
+  /** Logger to report information */
+  logger: BaseLogger;
+  /** Octokit instance to use to perform requests */
+  octokit: InstanceType<typeof ProbotOctokit>;
+  /** Targeted repository */
+  repo: { owner: string; repo: string };
+  /** Application ID */
+  appId?: string | number;
+}
+
+/**
+ * Cascading class implementation for Probot framework
+ */
+export class CascadingProbot extends Cascading {
+
+  /**
+   * Default configuration
+   */
+  public static readonly DEFAULT_CONFIGURATION: CascadingConfiguration = {
+    ignoredPatterns: [],
+    defaultBranch: '',
+    cascadingBranchesPattern: '^releases?/\\d+\\.\\d+',
+    versionCapturePattern: '/((?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)(?:\\.0-[^ ]+)?)$',
+    bypassReviewers: false
+  };
+
+  /** List of possible configuration files paths */
+  public static readonly CONFIGURATION_FILES = ['.github/cascadingrc.json', '.github/.cascadingrc.json', 'cascadingrc.json', '.cascadingrc.json'];
+
+  constructor(protected options: CascadingProbotOptions) {
+    super(options.logger);
+  }
+
+  /** @inheritdoc */
+  protected async loadConfiguration(currentBranch?: string) {
+    this.logger.debug('Load configuration');
+    let remoteConfig: Partial<CascadingConfiguration> = {};
+
+    const configFileResponses = await Promise.all(CascadingProbot.CONFIGURATION_FILES
+      .map((configFile) => this.options.octokit.repos.getContent({
+        ...this.options.repo,
+        path: configFile,
+        ref: currentBranch && `refs/heads/${currentBranch}`
+      }).catch(() => null)));
+
+    const configFileValidResponses = configFileResponses
+      .filter((res) => res && Array.isArray(res.data) && res.data.length > 0);
+
+    const configFileResponse = configFileValidResponses[0];
+
+    if (configFileValidResponses.length > 1) {
+      this.logger.warn(`Several configuration files have been found (${configFileValidResponses.map((c) => Array.isArray(c!.data) && c!.data[0].path).join(', ')}). The files will be ignored.`);
```

Comment:
not all the files will be ignored 
the first one is taken into account
maybe we should remove it from this list

Reply:
Actually all of are ignored, it is done on purpose to not define priority